src/paradigma/heart_rate/tfd.py

```diff
@@ -0,0 +1,553 @@
+    elif win_type in ['hamm', 'hamming']:
+        win = signal.windows.hamming(win_length)
+    elif win_type in ['hann', 'hanning']:
+        win = signal.windows.hann(win_length)
+    elif win_type == 'gauss':
+        win = signal.windows.gaussian(win_length, std=win_param if win_param else 0.4)
+    elif win_type == 'cosh':
+        win_hlf = win_length // 2
+        if not win_param:
+            win_param = 0.01
+        win = np.array([np.cosh(m) ** (-2 * win_param) for m in range(-win_hlf, win_hlf+1)])
+        win = np.fft.fftshift(win)
+    else:
+        raise ValueError(f"Unknown window type {win_type}")
+    
+    # If dft_window is True, return the DFT of the window
+    if dft_window:
+        win = np.fft.fft(np.roll(win, win_length // 2))
+        win = np.roll(win, -win_length // 2)
+    
+    return win
+
+def shift_window(w):
+    """
+    Shift the window so that positive indices appear first.
+    
+    Parameters:
+    -----------
+    w : ndarray
+        Window to be shifted.
+    
+    Returns:
+    --------
+    w_shifted : ndarray
+        Shifted window with positive indices first.
+    """
+    N = len(w)
+    return np.roll(w, N // 2)
+
+def pad_window(w, Npad):
+    """
+    Zero-pad the window to a specified length.
+    
+    Parameters:
+    -----------
+    w : ndarray
+        The original window.
+    Npad : int
+        Length to zero-pad the window to.
+    
+    Returns:
+    --------
+    w_pad : ndarray
+        Zero-padded window of length Npad.
+    
+    Raises:
+    -------
+    ValueError:
+        If Npad is less than the original window length.
+    """
+    N = len(w)
+    w_pad = np.zeros(Npad)
+    Nh = N // 2
+    
+    if Npad < N:
+        raise ValueError("Npad must be greater than or equal to the window length")
+
+    if N == Npad:
+        return w
+    
+    if N % 2 == 1:  # For odd N
+        w_pad[:Nh+1] = w[:Nh+1]
+        w_pad[-Nh:] = w[-Nh:]
+    else:  # For even N
+        w_pad[:Nh] = w[:Nh]
+        w_pad[Nh] = w[Nh] / 2
+        w_pad[-Nh:] = w[-Nh:]
+        w_pad[-Nh] = w[Nh] / 2
+    
+    return w_pad
+
+def compute_tfd(N, Nh, tfd):
+    """
+    Finalizes the time-frequency distribution computation.
+
+    Parameters:
+    -----------
+    N : int
+        Size of the TFD.
+    Nh : int
+        Half-length parameter.
+    tfd : ndarray
+        Time-frequency distribution to be finalized.
+
+    Returns:
+    --------
+    tfd : ndarray
+        Final computed TFD (N,N).
+    """
+    m = np.arange(0, Nh)  # m = 0:(Nh-1)
+    mb = np.arange(1, Nh)  # mb = 1:(Nh-1)
+
+    m_real = np.arange(Nh) # m_real = 0:(Nh-1) --> is this the same as m? In matlab it is different because of 1-based indexing
+    m_imag = np.arange(Nh, N) # m_imag = Nh:(N-1)
+
+    for n in range(0, N-1, 2):  # for n=0:2:N-2
+        R_even_half = np.complex128(tfd[n, :Nh]) + 1j * np.complex128(tfd[n, Nh:])
+        R_odd_half = np.complex128(tfd[n+1, :Nh]) + 1j * np.complex128(tfd[n+1, Nh:])
```

Comment:
```suggestion
        R_even_half = np.complex128(tfd[n, :Nh]) + j * np.complex128(tfd[n, Nh:])
        R_odd_half = np.complex128(tfd[n+1, :Nh]) + j * np.complex128(tfd[n+1, Nh:])
```

Reply:
1j is complex unit while j causes an error since it is not default in python